src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -223,22 +223,21 @@ class HuiMapCard extends LitElement implements LovelaceCard {
     });
   }
 
-  private _unsubscribeHistoryTimeWindow() {
+  private async _unsubscribeHistoryTimeWindow() {
     if (!this._subscribed) {
       return;
     }
-    this._subscribed.then((unsubscribe) => {
-      if (unsubscribe) {
-        unsubscribe();
-      }
-      this._subscribed = undefined;
-    });
+    const unsubscribe = await this._subscribed;
+    if (unsubscribe) {
+      unsubscribe();
+    }
+    this._subscribed = undefined;
   }
 
-  protected updated(changedProps: PropertyValues): void {
+  protected async updated(changedProps: PropertyValues) {
```

Comment:
This needs a different solution per https://github.com/home-assistant/frontend/pull/15217#discussion_r1088325647